File: ios/App/App/FirebaseNativePlugin.swift
import Foundation
import Capacitor
import FirebaseCore

@objc(FirebaseNativePlugin)
public class FirebaseNativePlugin: CAPPlugin {
    @objc func initialize(_ call: CAPPluginCall) {
        guard let config = call.getObject("config") else {
            call.reject("config is required")
            return
        }

        guard let apiKey = config["apiKey"] as? String,
              let appId = config["appId"] as? String,
              let projectId = config["projectId"] as? String else {
            call.reject("config.apiKey, config.appId, and config.projectId are required")
            return
        }

        if FirebaseApp.app() != nil {
            call.resolve(["status": "already-initialized"])
            return
        }

        let messagingSenderId = (config["messagingSenderId"] as? String)?.trimmingCharacters(in: .whitespacesAndNewlines)
        let storageBucket = (config["storageBucket"] as? String)?.trimmingCharacters(in: .whitespacesAndNewlines)
        let databaseURL = (config["databaseURL"] as? String)?.trimmingCharacters(in: .whitespacesAndNewlines)

        let gcmSender = (messagingSenderId?.isEmpty == false) ? messagingSenderId! : "placeholder-sender"
        let options = FirebaseOptions(googleAppID: appId, gcmSenderID: gcmSender)
        options.apiKey = apiKey
        options.projectID = projectId

        if let storageBucket, !storageBucket.isEmpty {
            options.storageBucket = storageBucket
        }
        if let databaseURL, !databaseURL.isEmpty {
            options.databaseURL = databaseURL
        }

        FirebaseApp.configure(options: options)
        call.resolve(["status": "initialized"])
    }
}

File: android/app/src/main/java/com/materialguardian/app/FirebaseNativePlugin.java
package com.materialguardian.app;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

import com.google.firebase.FirebaseApp;
import com.google.firebase.FirebaseOptions;

@CapacitorPlugin(name = "FirebaseNative")
public class FirebaseNativePlugin extends Plugin {

    @PluginMethod
    public void initialize(PluginCall call) {
        JSObject cfg = call.getObject("config");
        if (cfg == null) {
            call.reject("Missing 'config' object");
            return;
        }

        String appId = cfg.optString("appId", null);
        String apiKey = cfg.optString("apiKey", null);
        String projectId = cfg.optString("projectId", null);

        if (appId == null || apiKey == null || projectId == null) {
            call.reject("config must include appId, apiKey, projectId");
            return;
        }

        // Only initialize once
        if (FirebaseApp.getApps(getContext()).isEmpty()) {
            FirebaseOptions options = new FirebaseOptions.Builder()
                    .setApplicationId(appId)
                    .setApiKey(apiKey)
                    .setProjectId(projectId)
                    .setStorageBucket(cfg.optString("storageBucket", null))
                    .setGcmSenderId(cfg.optString("messagingSenderId", null))
                    .build();

            FirebaseApp.initializeApp(getContext(), options);
        }

        JSObject ret = new JSObject();
        ret.put("initialized", true);
        call.resolve(ret);
    }
}
